src/platform/plugins/shared/dashboard/public/dashboard_renderer/dashboard_renderer.tsx

```diff
@@ -131,15 +130,15 @@ export function DashboardRenderer({
         className="dashboardContainer"
         data-test-subj="dashboardContainer"
         css={styles.renderer}
-        ref={(e) => (dashboardContainerRef.current = e)}
+        ref={dashboardInternalApi.setDashboardContainerRef}
```

Comment:
This will cause problems. `dashboardInternalApi` is loaded async and may not be available when ref is provided. To avoid race conditions, how about

```
ref={(e) => {
  if (dashboardInternalApi) {
    dashboardInternalApi.setDashboardContainerRef(dashboardInternalApi);
    return;
  }

  dashboardContainerRef.current = e;
}}
```

Then when dashboardInternalApi is set, if `dashboardContainerRef.current` call setDashboardContainerRef